src/shared/stores/streamEditor.tsx

```diff
@@ -439,215 +434,225 @@ export const useStreamEditorStore = create<Actions & State>((set, get) => {
+                await toastedOperations(operations, async (next, refresh) => {
+                    if (!transientStreamId && !streamId) {
+                        throw new DraftValidationError('streamId', 'is required')
                     }
-                }
 
-                stream = await (async () => {
-                    client = await getTransactionalClient()
+                    if (streamId) {
+                        /**
+                         * Assign a draft to a stream id for later re-use. Based
+                         * on this information we can recycle drafts that haven't
+                         * been abandoned.
+                         */
+                        set((store) =>
+                            produce(store, ({ streamDraftMapping }) => {
+                                streamDraftMapping[streamId] = draftId
+                            }),
+                        )
+                    }
 
-                    const address = await client.getAddress()
+                    if (shouldUpdateMetadata && streamId) {
+                        updateOperation.action = getOpenStreamLink(streamId)
 
-                    await requirePositiveBalance(address)
+                        refresh()
+                    }
 
                     if (transientStreamId) {
-                        return client.createStream({
-                            id: transientStreamId,
-                            ...metadata,
+                        client = await getTransactionalClient({
+                            passiveNetworkCheck: true,
                         })
-                    }
 
-                    if (!streamId) {
-                        throw new DraftValidationError('streamId', 'is invalid')
-                    }
+                        try {
+                            if (await client.getStream(transientStreamId)) {
+                                throw new DraftValidationError(
+                                    'streamId',
+                                    'already exists, please try a different one',
+                                )
+                            }
+                        } catch (e) {
+                            if (e instanceof DraftValidationError) {
+                                throw e
+                            }
 
-                    if (metadataChanged) {
-                        return client.updateStream({
-                            ...metadata,
-                            id: streamId,
-                        })
+                            if (
+                                !isMessagedObject(e) ||
+                                !/stream not found/i.test(e.message)
+                            ) {
+                                throw new DraftValidationError(
+                                    'streamId',
+                                    'failed to verify uniqueness',
+                                )
```

Comment:
If it's an unknown error we do generate a validation error on the `streamId` field saying so.